### ISSUE TITLE
Nonfree components added

### ISSUE THREAD

**Issue by IzzySoft** [MahmoudMabrok/QuranyApp#9]
The last version (v1.6.11) added nonfree components, namely GMS and Firebase. Was that intentional – or (hopefully) just a mistake that can be reverted?

**Comment by MahmoudMabrok** [comment]
I have added Firebase as Backend to store users score and make scoreboard and get rank around the world.
I have been tried to make it with PHP but not succeded to make it.

**Comment by MahmoudMabrok** [comment]
but Firebase is free.

**Comment by IzzySoft** [comment]
> but Firebase is free

It's not OPEN. And it's on Google servers, thus potentially tracking your users (as it's not open, you can't check the code to make sure – especially not what happens server-side – and you are not free to use it with your own server). Further, it requires a network connection to work, making it harder to use your app while being offline.

**Comment by MahmoudMabrok** [comment]
you can use the app offline, need connections only to upload your score.
Later I will work to use an alternative to Firebase.

**Comment by IzzySoft** [comment]
Thanks @MahmoudMabrok – that's good to know, both of it 😃 It always hurts me seeing a good open source project getting tainted by proprietary libraries.

Besides: If you wish you can [link here](https://apt.izzysoft.de/fdroid/index/apk/education.mahmoud.quranyapp) for an alternative installation source. Can be used with the [F-Droid](https://f-droid.org/) client. Updates are available there within 24h of your placing them here at Github (and the client will inform its users). And for distraction-free viewing of the video, may I recommend [this link](https://invidio.us/watch?v=UosZoX5lgAs)? 😄

Thanks for all, and keep up the good work!

**Comment by MahmoudMabrok** [comment]
Thanks for your great efforts.

**Comment by IzzySoft** [comment]
Thanks for adding the link! ❤️

**Comment by MahmoudMabrok** [comment]
I have been removed Firebase from my App @IzzySoft

**Comment by IzzySoft** [comment]
That's good news! I saw the new APK was just uploaded an hour ago, so it should be fetched tomorrow to my repo and I can verify. Oh, correction: It was already removed 6 days ago, and my scanner confirms it – congrats and a big thanks! Do you want to go for the official F-Droid repo then? Need assistance with that?

**Comment by MahmoudMabrok** [comment]
what I need to go to  official F-Droid repo?

**Comment by IzzySoft** [comment]
First step would be filing an [RFP](https://gitlab.com/fdroid/rfp/issues) (Request For Packaging). Giving the proper details, your repo would be scanned by a bot which then e.g. confirms there are no proprietary libraries, precompiled JARs etc. used. Metadata would be reviewed (description, links), you'd be asked if you could setup fastlane or triple-t structures for screenshots and localized descriptions (which is optional, but strongly recommended, see [here](https://f-droid.org/en/docs/All_About_Descriptions_Graphics_and_Screenshots/#in-the-apps-source-repository) for details; you wouldn't need to use those tools but using the directory structure they define, so screenshots etc. can be picked up automatically). Then someone of the team would try to build your app – and if that succeeds, it will be listed with the next index build. For a more detailed description, you can see my article [F-Droid for advanced users and developers](https://android.izzysoft.de/articles/named/fdroid-intro-2).

Disclosure: I'm one of the maintainers there, so I can assist you. I would e.g. prepare the metadata template for you (most of it is already setup in my repo, so that's just a little copy-pasting).

**Comment by MahmoudMabrok** [comment]
Thanks for  your help, I will do that later ISA as I have exams these days

**Comment by IzzySoft** [comment]
Take your time, first things first. Just ping me if you need assistance. Good luck for your exams!

**Comment by MahmoudMabrok** [comment]
Thanks for the support.

**Comment by IzzySoft** [comment]
@MahmoudMabrok where did the APK go to? My updater just reported it cannot find it anymore. Without you providing it, I cannot update it in my repo.

**Comment by MahmoudMabrok** [comment]
I am sorry for that, but I have started to add link of Google Drive as adding apk repo will be larger size and add more load for cloning repo.
The link is https://drive.google.com/file/d/1iU-WgufHMAELKOmClhW2Z18Miyef4f4v/view

**Comment by IzzySoft** [comment]
Sorry to disappoint you, but my updater doesn't work with that. Why not [tagging your releases](https://github.com/MahmoudMabrok/QuranyApp/releases) and attaching the APK there? I'm fully with you that "inside the repo" isn't the best place, but IMHO `releases/` is :wink:

**Comment by IzzySoft** [comment]
Thanks for the tag, @MahmoudMabrok – will you also attach the `.apk` to it, and keep this up for future releases? Currently I had to disable auto-updates for your app (and it's stuck at v2.1.3 on my end because of that), but I'd really prefer having them active and fetch new versions as soon as you released them :smiley:

**Comment by MahmoudMabrok** [comment]
I added it as an asset to release `sources.zip`  it contains `APK`.
Is `aab` -android bundle - supported with your platform?

**Comment by IzzySoft** [comment]
Nope, it's not, sorry. And neither are ZIP files; so why zipping the APK if you can attach it as-is? Besides, I don't even see the `sources.zip`…

**Comment by MahmoudMabrok** [comment]
I uploaded the apk but Github added it as `source code.zip`

**Comment by MahmoudMabrok** [comment]
![aa](https://user-images.githubusercontent.com/13488900/61201313-66440980-a6e4-11e9-9599-357f8667314c.JPG)

**Comment by IzzySoft** [comment]
Aren't those two the archives Github adds by default (zipping/tar-gnuzipping the entire tree)? Could you try attaching the APK again?

**Comment by MahmoudMabrok** [comment]
I will try it later as I am busy now with my graduation project.
I AM SORRY.

**Comment by IzzySoft** [comment]
Sure, first things first (all the best for your graduation!). Just ping me here when I shall check again. Thanks!

**Comment by IzzySoft** [comment]
Can we congratulate to your graduation? And any chance for the APK then? :smiley:

**Comment by MahmoudMabrok** [comment]
Yes you can :smiley:, I am sorry for not maintaining the project.
but i do not work in app nowadays.
what you need so i can prepare it?

**Comment by IzzySoft** [comment]
Then I hereby do :tada:

I'm no dev, so I cannot build the app myself. If you have the latest APK and could attach it to its corresponding release, that would be great. After that, if you no longer develop/maintain the app, maybe add a corresponding note to the Readme, and archive the project?

**Comment by MahmoudMabrok** [comment]
I am sorry for being late to response to you dir. 
what you exactly you want from me?.
@IzzySoft

**Comment by IzzySoft** [comment]
Uh, lost track – indeed quite a while ago… Could you attach the last APK to the currently only left tag [here](https://github.com/MahmoudMabrok/QuranyApp/releases/tag/v2.5.1)? There's no APK anymore. And us non-devs cannot compile ourselves. Thanks in advance, @MahmoudMabrok!

**Comment by MahmoudMabrok** [comment]
I will try to add APK tomorow ISA, if i could not i will make a folder and put apk inside it. @IzzySoft

**Comment by IzzySoft** [comment]
Yeah, ugly as it is, inside the repo would do as well. Thanks!

**Comment by IzzySoft** [comment]
Just saw the APK. Would it be possible to use the (signed) release build instead of the debug one? Just if it's not too much trouble. Thanks!
